.github/workflows/build.yml

```
@@ -0,0 +1,142 @@
+        os: [ ubuntu-latest, windows-latest, macos-latest ]
+      fail-fast: false
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - name: Checkout the latest code
+      uses: actions/checkout@v3
+
+    - name: Setup Node
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ env.NODE_VERSION }}
+
+    - name: Setup Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ env.PYTHON_VERSION }}
+
+    - name: Setup Git Submodule
+      run: |
+        git submodule init
+        git submodule update
+
+    - name: Check Pulsar Version
+      if: ${{ matrix.os != 'windows-latest' }}
+      run: sed -i -e "s/[0-9]*-dev/`date -u +%Y%m%d%H`/g" package.json
+
+    - name: Check Pulsar Version - Windows
+      if: ${{ matrix.os == 'windows-latest' }}
+      run: (Get-Content package.json) -replace '[0-9]*-dev', (date -u +%Y%m%d%H) | Set-Content -Path package.json
+
+    - name: Install Pulsar Dependencies
+      uses: nick-fields/retry@v2
+      with:
+        timeout_minutes: 30
+        max_attempts: 3
+        retry_on: error
+        command: yarn install --ignore-engines
+        on_retry_command: rm -R node_modules
+
+    - name: Build Pulsar
+      uses: nick-fields/retry@v2
+      with:
+        timeout_minutes: 30
+        max_attempts: 3
+        retry_on: error
+        command: |
+          yarn build
+          yarn run build:apm
+
+    # macOS Signing Stuff
+    - name: Build Pulsar Binaries (macOS)
+      if: ${{ matrix.os == 'macos-latest' }}
+      # env:
+      #   CSC_LINK: ${{ secrets.CSC_LINK }}
+      #   CSC_KEY_PASSWORD: ${{ secrets.CSC_KEY_PASSWORD }}
+      #   APPLEID: ${{ secrets.APPLEID }}
+      #   APPLEID_PASSWORD: ${{ secrets.APPLEID_PASSWORD }}
+      #   TEAM_ID: ${{ secrets.TEAM_ID }}
+      uses: nick-fields/retry@v2
+      with:
+        timeout_minutes: 30
+        max_attempts: 3
+        retry_on: error
+        command: yarn dist
+
+    - name: Build Pulsar Binaries
+      if: ${{ matrix.os != 'macos-latest' }}
+      uses: nick-fields/retry@v2
+      with:
+        timeout_minutes: 30
+        max_attempts: 3
+        retry_on: error
+        command: yarn dist
+
+    - name: Upload Binary Artifacts
+      uses: actions/upload-artifact@v3
+      with:
+        name: ${{ matrix.os }} Binaries
+        path: ./binaries/*
+
+    - name: Test Binary - Linux
+      if: ${{ (matrix.os == 'ubuntu-latest') && env.RUN_LINUX_VT }}
+      run: |
+        rm -R node_modules/electron; yarn install --check-files
+        ./binaries/*AppImage --appimage-extract
+        export BINARY_NAME='squashfs-root/pulsar'
+        mkdir -p ./tests/videos
+        Xvfb -screen 0 1024x768x24+32 :99 & nohup ffmpeg -video_size 1024x768 -f x11grab -i :99.0 ./tests/videos/out.mpg & DISPLAY=:99 PLAYWRIGHT_JUNIT_OUTPUT_NAME=report.xml npx playwright test --reporter=junit,list
+
+    - name: Test Binary - Windows
+      if: matrix.os == 'windows-latest' && env.RUN_WINDOWS_VT == true
```

Broader Windows check

https://docs.github.com/en/actions/learn-github-actions/contexts#runner-context

```suggestion
      if: runner.os == 'Windows' && env.RUN_WINDOWS_VT == true
```